test/spark/run-client-test.sh

```diff
@@ -1,44 +1,121 @@
 #!/bin/bash -aux
```

Comment:
As a hater of shell scripts, can we add `set -o pipefail` here, too?

Reply:
done